### PR TITLE
-> v1.0.0

### PR DESCRIPTION
**Pull request by mhowlett**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,38 @@
...
+1.0.0 is a major update of the API. Feature highlights:
+
+- Inherits all of the new features in librdkafka [v1.0.0](https://github.com/edenhill/librdkafka/releases/tag/v1.0.0)
+- General improvements to client classes:
+  - Strongly typed configuration.
+  - Construction is via builder classes:
+    - Allows/enforces that event handlers are specified at construction time.
+    - More extensible.
+  - Header support.
+  - New Message class abstraction and (many) flow on changes.
```
Comment (edenhill) [inline CHANGELOG.md]: "flow on changes" ?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,38 @@
...
+    - Non-blocking support for async serializers.
+    - Very flexible:
+      - e.g. can be easily extended to support header serialization.
+  - Capability to specify custom timestamps when producing messages. 
+  - Message persistence status support.
+  - Renamed ProduceAsync variants with a callback to BeginProduce.
+- Consumer improvements:
+  - A new rebalance API.
+  - New deserialization API analogous to the new serialization API.
+  - PartitionEOF notification is via ConsumeResult, not events.
```
Comment (edenhill) [inline CHANGELOG.md]: And now needs to be explicitly enabled through EnablePartitionEOF (or whatever it is called)

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,38 @@
...
+- SchemaRegistry integration
```
Comment (edenhill) [inline CHANGELOG.md]: Confluent Schema Registry

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,38 @@
+# 1.0.0
+
+1.0.0 is a major update of the API. Feature highlights:
```
Comment (edenhill) [inline CHANGELOG.md]: Point out that the API changes are breaking, non-compatible with previous versions.

Feature highlights should go on its own line.

Comment (bartelink) [reply CHANGELOG.md]: Consider using KeepAChangeLog format e.g. https://github.com/jet/Jet.ConfluentKafka.FSharp/blob/v1/CHANGELOG.md (follow link within - its really easy and really useful)

File: CHANGELOG.md (hunk 1 above)
Comment (mhowlett) [reply CHANGELOG.md]: changed to 'and related changes'.

File: CHANGELOG.md (hunk 4 above)
Comment (mhowlett) [reply CHANGELOG.md]: thanks @bartelink 
v1.0 is tricky because it's essentially a complete re-write. Following 1.0, this sounds like a great idea.

**Comment by mhowlett** [comment]
closing in favor of #907

**Pull request closed without merging**